### PR TITLE
Add initial copy of OpAMP spec

### PR DESCRIPTION
**Pull request by tigrannajaryan**
The README.md file is originally Apache 2.0-Licensed. It is an exact copy of
https://github.com/signalfx/opamp-spec/blob/main/README.md.

According to Apache License 2.0 requirements and OpenTelemetry guidelines
(https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md) we
are going to retain the original copyright notice.

As recommended by the guidelines we will add "Copyright The OpenTelemetry Authors"
notice once we make any changes to this file.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@djaglowski @codeboten please review.

**Review by djaglowski — APPROVED** [review]
Approving w/o detailed review because this PR is bringing over the spec written elsewhere, and because we expect substantial iterate on this. I'll raise my questions later as issues or in the working group.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,1743 @@
...
+
+<h1 id="communication-model">Communication Model</h1>
+
+
+The OpAMP Server manages Agents that implement the client-side of OpAMP protocol. The Agents can optionally send their own telemetry to an OTLP destination when directed so by the OpAMP Server. The Agents likely also connect to other destinations, where they send the data they collect:
+
+
+```
+        ┌────────────┬────────┐           ┌─────────┐
+        │            │ OpAMP  │  OpAMP    │ OpAMP   │
```
Comment (pmm-sumo) [inline README.md:50]: This is less than a nit but perhaps to emphasise the protocol used and for consistency we could use something along following
```suggestion
        │            │ OpAMP  │  OpAMP/WS │ OpAMP   │
```

**Comment by pmm-sumo** [comment]
The specification lists a rich set of capabilities, which as I believe will be prioritised and iteratively added. I am wondering what should be the flow when agent is not able to handle some changes (e.g. is not able to handle addons yet). Perhaps special status codes to indicate this type of problem could be used (so server could understand if something really failed or it was not possible to apply it)?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,1743 @@
...
+
+<h3 id="agentinstallstatus-message">AgentInstallStatus Message</h3>
+
+
+This message contains the status of the last agent package install status performed by the agent and has the following structure:
+
+
+```protobuf
+message AgentInstallStatus {
+    bytes server_offered_hash = 1;
```
Comment (pmm-sumo) [inline README.md:621]: I think we are missing here the additional version information which we discussed in the Google Doc :) 
```suggestion
    bytes server_offered_version = 1;
    bytes server_offered_hash = 2;
```
(and so on)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,1743 @@
...
+<h3 id="agentaddonstatus-message">AgentAddonStatus Message</h3>
+
+
+The AgentAddonStatus has the following structure:
+
+
+```protobuf
+message AgentAddonStatus {
+    string name = 1;
+    bytes agent_has_hash = 2;
```
Comment (pmm-sumo) [inline README.md:559]: Perhaps we could add `string server_offered_version` here as well?

**Review by pmm-sumo — APPROVED** [review]
Just as @djaglowski, I am good with merging the spec as it is now and continue discussion using separate issues/PR's

**Review by codeboten — COMMENTED** [review]
Just one question does this fall under the category of a donation that must be reviewed by the TC? https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#donations

Or has a review already been done and is documented somewhere?

**Comment by tigrannajaryan** [comment]
> The specification lists a rich set of capabilities, which as I believe will be prioritised and iteratively added. I am wondering what should be the flow when agent is not able to handle some changes (e.g. is not able to handle addons yet). Perhaps special status codes to indicate this type of problem could be used (so server could understand if something really failed or it was not possible to apply it)?

Added an issue to clarify it: https://github.com/open-telemetry/opamp-spec/issues/7

File: README.md (hunk 2 above)
Comment (tigrannajaryan) [reply README.md:621]: Yes, I will add changes as separate PRs.

File: README.md (hunk 3 above)
Comment (tigrannajaryan) [reply README.md:559]: Created an issue to decide https://github.com/open-telemetry/opamp-spec/issues/8

**Comment by tigrannajaryan** [comment]
> Just one question does this fall under the category of a donation that must be reviewed by the TC? https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#donations
> 
> Or has a review already been done and is documented somewhere?

AFAIK from legal standpoint it is not a donation. We are merely using an existing Apache-licensed content as a starting point.

I don't think we need a review from TC since we are not bringing an entire existing product. It is relatively small bit that we take as a draft and will work on as a Workgroup. I believe this is what we should typically expect from a Workgroup.

**Pull request merged**